base/pipelines/files/model-deployment.yaml

```diff
@@ -0,0 +1,55 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: MODEL_NAME-MODEL_VERSION # CHANGE THIS
```

Comment:
I know this is just a placeholder, but it's not a valid name for an object in Kubernetes, because of the `_`s. See here for more info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names

For the edge apps, kustomize expects that the contents of the base definitions is valid, and will just prefix/suffix some things like names, rather than replace them. I think this would be a good assumption to keep, rather than treating these as mere templates.

When the ACM subscription tries to apply the kustomized resource to the target cluster, it will hit errors such as the following:

```
Error from server (Invalid): error when creating "acm/odh-edge/apps/bike-rental-app/": Deployment.apps "bike-rental-app-MODEL_NAME-MODEL_VERSION" is invalid: [metadata.name: Invalid value: "bike-rental-app-MODEL_NAME-MODEL_VERSION": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'), spec.template.spec.containers[0].image: Required value, spec.template.spec.containers[0].readinessProbe.httpGet.port: Invalid value: 0: must be between 1 and 65535, inclusive]
```

Reply:
I am also aware that K8s will not allow "_" which is why I chose it with the aim of a template not being used as-is accidentaly. Can you elaborate on the kustomize bit about prefix/suffixing rather than replacing? I think using [`op: replace`](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/patches/) and replacing the placeholders would be the best approach.